core/src/main/java/com/novoda/merlin/service/Ping.java

```diff
@@ -14,12 +15,16 @@
 
     public boolean doSynchronousPing() {
         MerlinLog.d("Pinging : " + hostAddress);
-        int responseCode = responseCodeFetcher.from(hostAddress);
-        MerlinLog.d("Got response : " + responseCode);
-        return isSuccess(responseCode);
-    }
+        try {
+            responseCodeFetcher.from(hostAddress);
+        } catch (RequestException e) {
+            if (e.causedByIO()) {
```

Comment:
but I thought this bit of code here checks if you have a _real_ internet connection not just that you are connected to a network.

i.e. what is the behaviour now if you are in an internet cafe and have not signed in? (it should return false)

@ouchadam is this the place that should be doing what I said above?